doc/amoc_coordinator.md

```diff
@@ -0,0 +1,170 @@
+# amoc_coordinator
+
+## Module
+
+`amoc_coordinator` 
+
+## Description
+
+This module allows to synchronize the users, and act on groups of them.
+
+The coordinator reacts to new users showing up in a system, according to the `Coordination Plan`. 
+The `Coordination Plan` consists of `Coordination Items`, and each of them is defined as one of the following: `{NumberOfUsers, CoordinationActions}`.
+ - When the `Number Of Users` is set to `all`, then only `Coordination Actions` with the arities `/1, /2` are handled. 
+ The `Coordination Items` with `all` are triggered by the `timeout` event. 
+ - When the `Number Of Users` is set to a positive integer, all `Coordination Actions` with arities `/1, /2` and `/3` are handled.
+
+The timeout timer is reset by calling the `add` function.
+A new batch size is set in the `NumberOfUsers`. Each user in the batch calls the `add` function registering to the coordinator and triggering the `Coordination Plan`.
+If more then one of the `Coordination Items` matching the `NumberOfUsers` is triggered, each of them will be passed a respective number of users.
+For example if the `Coordination Plan` is `[{2, Act1}, {3, Act2}]` then on the 6th user calling `add`, `Act1` will be called with 2 users passed and `Act2` wil be called with 3 users passed.
+
+`Coordination Actions` may be one of the following:
+ - `fun(Event) -> any()` - this type of action does not care about particular users, but only about the number of them;
+ - `fun(Event, ListOfUsersData) -> any()` - this type of action gets `ListOfUsersData` which is a list of tuples `{Pid, Data}` with `Pid`s passed by users calling `amoc_coordinator:add/2` or `amoc_coordinator:add/3`;
+- `fun(Event, User1, User2) -> any()` - this type of action gets `distinct pairs` from the batch of users `User1` and `User2` which are tuples `{Pid, Data}` with `Pid`s passed by users calling `amoc_coordinator:add/2` or `amoc_coordinator:add/3`;
+
+`Coordination Actions` are executed in a reversed order in relation to their declaration in the `Coordination Plan`.
```

Comment:
this is not true. the order of actions in one item is guaranteed (direct order, not reversed). the order of actions from different coordination items is not guaranteed at all (casted and executed in different processes). timeout/reset/stop events are not casted but ‘called’, so for the timeout/reset/stop evenst it’s guaranteed that all the coordination items with `all` are executed after all numbered coordination items are done.